qubes/tests/__init__.py

```diff
@@ -1328,33 +1328,54 @@ def prepare_hvm_system_linux(self, vm, init_script, extra_files=None):
 
     def create_bootable_iso(self):
         """Create simple bootable ISO image.
-        Type 'poweroff' to it to terminate that VM.
+        Type 'halt' to it to terminate that VM.
         """
-        isolinux_cfg = (
-            'prompt 1\n'
-            'label poweroff\n'
-            '   kernel poweroff.c32\n'
-        )
         output_fd, output_path = tempfile.mkstemp('.iso')
         with tempfile.TemporaryDirectory() as tmp_dir:
+            f = os.open(tmp_dir.name)
             try:
-                shutil.copy('/usr/share/syslinux/isolinux.bin', tmp_dir)
-                shutil.copy('/usr/share/syslinux/ldlinux.c32', tmp_dir)
-                shutil.copy('/usr/share/syslinux/poweroff.c32', tmp_dir)
-                with open(os.path.join(tmp_dir, 'isolinux.cfg'), 'w') as cfg:
-                    cfg.write(isolinux_cfg)
-                subprocess.check_call(['genisoimage', '-o', output_path,
-                                       '-c', 'boot.cat',
-                                       '-b', 'isolinux.bin',
+                os.mkdir('os', mode=0o755, dir_fd=f)
+                os.mkdir('iso', mode=0o755, dir_fd=f)
+                os.mkdir('os/images', mode=0o755, dir_fd=f)
+                os.mkdir('os/EFI', mode=0o755, dir_fd=f)
+                os.mkdir('os/EFI/BOOT', mode=0o755, dir_fd=f)
+                subprocess.check_call(['sudo', 'grub2-mkimage',
+                                       '-O', 'i386-pc-eltorito',
+                                       '-d', '/usr/lib/grub/i386-pc',
+                                       '-p', '/boot/grub2',
+                                       'iso9660', 'biosdisk', 'halt'])
+                graft_path = '.=' + os.path.join(tmp_dir, "os").replace("\\", "\\\\").replace("=", "\\=")
+                efiboot_img = os.path.join(tmp_dir, 'os/images/efiboot.img')
+                with open('os/EFI/BOOT', mode=0o644, dir_fd=f) as g:
+                    subprocess.check_call(['sudo', 'cat', '/boot/efi/EFI/qubes/grubx64.efi'], stdout=g)
```

Comment:
This file won't exists on a system installed in a legacy boot mode (like most instances in openQA). But also, do you really need it? A `grub2-mkimage` call similar to the above should give you what you need, no?

Reply:
I’ll need to figure it out (later).